core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -370,9 +381,27 @@ class LogManager(val logDirs: Array[File],
   }
 
   /**
-   *  Delete a log.
+   *  Delete logs marked for deletion.
    */
-  def deleteLog(topicAndPartition: TopicAndPartition) {
+  private def deleteLogs(): Unit = {
```

Comment:
@jjkoshy I think this is a race condition and has been an existing issue with Delete. 
Lets consider the following scenario : 
1) The kafka-log-retention thread runs and gets all the logs.
2) Now someone issues a deleteTopic request and the directory gets moved. The kafka-log-retention is not aware of this. 
3) At this exact same point the async deleteLog thread runs and deletes the logs for the topic.
4) Now when the kafka-log-retention runs, it will get an NPE. 

Is this the scenario you were hinting at?